server/api/users.js

```diff
@@ -26,30 +31,23 @@ router.put(
     const newName = req.body.name;
     const newPassword = req.body.password;
     const newEmail = req.body.email;
-
-    const currentName = req.user.name;
-    const currentEmail = req.user.email;
-    const currentPassword = req.user.password;
-
-    if (req.user == null || req.user == undefined) {
+    const nameResult = nameValidate(newName);
+    const emailResult = emailValidate(newEmail);
+    const passwordResult = passwordValidate(newPassword);
+    if (!req.use) {
```

Comment:
You have a typo here, should be req.user

Reply:
ok